src/actions/request.rs

```diff
@@ -215,9 +215,33 @@ impl Runnable for Request {
         });
 
         if let Some(&SetCookie(ref cookies)) = response.headers.get::<SetCookie>() {
-          if let Some(cookie) = cookies.iter().next() {
-            let value = String::from(cookie.split(";").next().unwrap());
-            context.insert("cookie".to_string(), Yaml::String(value));
+          let context_cookies = context.entry("cookies".to_string()).or_insert_with(|| Yaml::Array(Vec::new()));
+          if let Yaml::Array(array) = context_cookies {
```

Comment:
I think it could be cool to use a let Some here and avoid the previous `or_insert_with` call

Reply:
The goal is to ensure there is a value in the context before this value is filled/updated. Then I need to check it has the right type so that the Yaml::Array content can be accessed.

Reply:
Ok. 👍